backend/ring/ring.go

```diff
@@ -124,7 +114,7 @@ func (r *Ring) supervise() error {
 		defer mu.Unlock(context.Background())
 		for range r.wakeup {
 			if err := r.advance(); err != nil {
-				logger.WithError(err).Error("couldn't get the next ring item")
+				logger.WithError(err).Error("supervisor error")
```

Comment:
I changed the error message so that developers would know the failure was in the supervisor goroutine, as opposed to the code that gets executed synchronously when calling the Ring's Next method. Not terribly attached to this particular change, I can change it back if you think it's better the other way.

Reply:
A compromise might be to have it be descriptive in both dimensions.

> supervisor: couldn't get the next ring item